src/main/java/org/stellar/sdk/SetTrustlineFlagsOperation.java

```diff
@@ -0,0 +1,171 @@
+package org.stellar.sdk;
+
+import com.google.common.base.Objects;
+import org.stellar.sdk.xdr.*;
+
+import java.util.EnumSet;
+
+/**
+ *
+ * Represents a Set Trustline Flags operation.
+ * @see <a href="https://www.stellar.org/developers/learn/concepts/list-of-operations.html" target="_blank">List of Operations</a>
+ */
+public class SetTrustlineFlagsOperation extends Operation {
+  private final String trustor;
+  private final AssetTypeCreditAlphaNum asset;
+  private final EnumSet<TrustLineFlags> clearFlags;
+  private final EnumSet<TrustLineFlags> setFlags;
+
+  private SetTrustlineFlagsOperation(String trustor, AssetTypeCreditAlphaNum asset, EnumSet<TrustLineFlags> clearFlags, EnumSet<TrustLineFlags> setFlags) {
+    this.trustor = trustor;
+    this.asset = asset;
+    this.clearFlags = clearFlags;
+    this.setFlags = setFlags;
+  }
+
+  /**
+   * The account owning of the trustline.
+   */
+  public String getTrustor() {
+    return trustor;
+  }
+
+  /**
+   * The asset of the trustline.
+   */
+  public Asset getAsset() {
+    return asset;
+  }
+
+  /**
+   * The flags to be set.
+   */
+  public EnumSet<TrustLineFlags> getSetFlags() {
+    return setFlags;
+  }
+
+  /**
+   * The flags to be cleared.
+   */
+  public EnumSet<TrustLineFlags> getClearFlags() {
+    return clearFlags;
+  }
+
+  private static Uint32 bitwiseOr(EnumSet<TrustLineFlags> set) {
+    int v = 0;
+    for (TrustLineFlags f : set) {
+      v |= f.getValue();
+    }
+    Uint32 combined = new Uint32();
+    combined.setUint32(v);
+    return combined;
+  }
+
+  @Override
+  org.stellar.sdk.xdr.Operation.OperationBody toOperationBody() {
+    SetTrustLineFlagsOp op = new SetTrustLineFlagsOp();
+
+    op.setTrustor(StrKey.encodeToXDRAccountId(this.trustor));
+    op.setAsset(asset.toXdr());
+    op.setClearFlags(bitwiseOr(clearFlags));
+    op.setSetFlags(bitwiseOr(setFlags));
```

Comment:
I think in the [newest version of XDR](https://github.com/stellar/go/blob/b21259c79556a2bf2515e612e879831e1f63fc4d/xdr/Stellar-transaction.x#L409) these fields are nullable which means this code will have to be changed. Are you planning to open a new PR or change it now?

Reply:
actually in the newest version of the xdr the fields are not pointers anymore, see: https://github.com/stellar/go/pull/3458

Reply:
OK, I wasn't aware of this.